src/utils/satelliteUtils.ts

```diff
@@ -1,17 +1,25 @@
 import { Satellite } from './../db/schema/satellite'
+import SatelliteInterface from '../utils/types/satelliteInterface'
 export default class SatelliteUtilities {
 
-    static async createSatelite(newSatelite: Record<string, string>): Promise<void> {
-        await Satellite.create(newSatelite)
-            .catch((err) => {
-                console.log(err)
-                throw 'error occured while putting new satellite'
-            })
-        return Promise.resolve()
+    static async createSatelite(newSatelite: SatelliteInterface): Promise<void> {
+        try {
+            await Satellite.create(newSatelite)
+        } catch (err) {
+            throw 'error occured while putting new satellite'
+        }
+    }
+
+    static getSatelliteById(id: string): Promise<SatelliteInterface | null> {
+        return Satellite.findById(id).exec();
+    }
+
+    static getSatelliteByName(name: string): Promise<SatelliteInterface | null> {
+        return Satellite.findOne({ officialName: name }).exec()
     }
 
     static getSateliteCount(): Promise<number> {
-        return Satellite.countDocuments()
+        return Satellite.countDocuments().exec()
```

Comment:
Remember that in large databases, esimatedDocumentCount is a significantly more performant option as it doesn't try to count the entire document list.  However, in this case countDocuments is fine, just keep it in mind.